qa/full-cluster-restart/src/javaRestTest/java/org/elasticsearch/upgrades/FullClusterRestartIT.java

```diff
@@ -268,7 +270,10 @@ public void testNewReplicas() throws Exception {
     }
 
     public void testSearchTimeSeriesMode() throws Exception {
-        assumeTrue("indexing time series indices changed in 8.2.0", getOldClusterVersion().onOrAfter(Version.V_8_2_0));
+
+        var originalClusterHasNewTimeSeriesIndexing = parseLegacyVersion(getOldClusterVersion()).map(v -> v.onOrAfter(Version.V_8_2_0))
```

Comment:
You may notice that the various `assumeTrue` still use a Version: this is because these tests are run twice, independently, once on the OLD cluster and once on the UPGRADED cluster, so they both need to check a condition against the _original_ cluster, in order to skip both.

Hopefully this is provisional: I'm talking to delivery to figure out a way to skip the OLD test (based on cluster features) _and if this happens_ also skip the same tests on the UPGRADED cluster. I have already an idea but it's "pure Java" and want to know if we can do something better with some gradle magic :) 

Once we have such a mechanism, I will revise these `assumeTrue(condition_on_old_cluster_version)` to be on (old) cluster features.

Reply:
This seems fine if there's a PR coming in the next week or so. Otherwise, what would you think of a wrapper method for these calls to `assume` that we could put a comment/TODO on?

e.g.

```
/** TODO ... */
public void assumeOriginalClusterCondition(String reason, boolean condition) {
    assumeTrue(reason, condition);
}
```

Reply:
Let me address this in a separate PR where I draft a possible approach